app/Filament/Resources/UserResource/Pages/ListUsers.php

```diff
@@ -150,14 +152,26 @@ public function table(Table $table): Table
                     ->visible((fn () => auth()->user()->can('user.*.restore'))),
                 SelectFilter::make('teams')
                     ->label('Team')
-                    ->relationship('teams', 'name')
+                    ->options($this->getTeamsOption())
+                    ->getSearchResultsUsing(fn (string $search) => Team::query()->where('name', 'like', '%' . $search . '%')->take(50)->pluck('name', 'id')->toArray())
+                    ->query(fn (Builder $query, array $data) => $this->teamFilter($query, $data))
                     ->multiple()
                     ->searchable()
                     ->preload(),
             ])
             ->defaultSort('name', 'asc');
     }
 
+    public function getTeamsOption(): array
+    {
+        $teams = Team::query()->take(50)->pluck('name', 'id')->toArray();
+
+        return [
+            'unassigned' => 'Unassigned',
+            ...$teams,
```

Comment:
It is expected that this options are grouped: https://filamentphp.com/docs/3.x/forms/fields/select#grouping-options